build.sbt

```diff
@@ -196,7 +192,20 @@ val sharedJavacOptions = List(
   }
 )
 
-val sharedSettings = sharedJavacOptions ++ List(
+val sharedScalacOptions = List(
+  scalacOptions ++= {
+    CrossVersion.partialVersion(scalaVersion.value) match {
+      case partialVersion if isScala211(partialVersion) =>
+        List("-target:jvm-1.8", "-Yrangepos", "-Xexperimental")
+      case partialVersion if isScala3(partialVersion) =>
```

Comment:
ha and very soon we can start detecting 3.3.x and add in unused warnings as well.